airflow/contrib/hooks/spark_submit_hook.py

```diff
@@ -220,6 +233,25 @@ def _build_command(self, application):
 
         return connection_cmd
 
+    def _build_track_driver_status_command(self):
+        """
+        Construct the command to poll the driver status.
+
+        :return: full command to be executed
+        """
+        connection_cmd = self._get_spark_binary_path()
+
+        # The url ot the spark master
+        connection_cmd += ["--master", self._connection['master']]
+
+        # The driver id so we can poll for its status
+        if self._driver_id:
+            connection_cmd += ["--status", self._driver_id]
+
```

Comment:
I understand there is already a first check in `_process_log` though I would also raise an exception here in the `else` case, something like
```
else
    raise AirflowException("Invalid status: attempted to poll driver status but no driver id is known. Giving up.")
```

Otherwise, in case of bug where the driver_id is not known, this function would silently create a weird to interpret error due to incorrect spark-submit process